### PR TITLE
Implement account_lines with new RPC framework

### LINKED ISSUE #550: Implement account_lines in new RPC system
Implement nextgen handler for `account_lines`.

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #550

### CODE REVIEW

File: src/rpc/ngHandlers/AccountLines.cpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    auto const accountID = RPC::accountFromStringStrict(input.account);
+    auto const accountLedgerObject = sharedPtrBackend_->fetchLedgerObject(
+        ripple::keylet::account(*accountID).key, lgrInfo.seq, yield);
+
+    if (not accountLedgerObject)
+        return Error{RPC::Status{
+            RPC::RippledError::rpcACT_NOT_FOUND, "accountNotFound"}};
+
+    auto const peerAccountID = input.peer
+        ? RPC::accountFromStringStrict(input.peer.value())
```
Comment (cindyyan317) [inline src/rpc/ngHandlers/AccountLines.cpp]: *(input.peer) maybe more efficient than value()  .

File: src/rpc/ngHandlers/AccountLines.h [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    struct LineResponse
+    {
+        std::string account;
+        std::string balance;
+        std::string currency;
+        std::string limit;
+        std::string limitPeer;
+        uint32_t qualityIn;
+        uint32_t qualityOut;
+        std::optional<bool> noRipple;
```
Comment (cindyyan317) [inline src/rpc/ngHandlers/AccountLines.h]: no need to be optional

File: src/rpc/ngHandlers/AccountLines.h [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    };
+
+    struct Input
+    {
+        std::string account;
+        std::optional<std::string> ledgerHash;
+        std::optional<uint32_t> ledgerIndex;
+        std::optional<std::string> peer;
+        std::optional<bool>
+            ignoreDefault;  // TODO: document
```
Comment (cindyyan317) [inline src/rpc/ngHandlers/AccountLines.h]: we can just set its default value instead using optional

**Review by cindyyan317 — DISMISSED** [review]
Just leave some minor things to check.

File: src/rpc/ngHandlers/AccountLines.cpp (hunk 1 above)
Comment (godexsoft) [reply src/rpc/ngHandlers/AccountLines.cpp]: because UB instead of exception? i don't think it's a huge performance boost but i'll change as you suggest 👍

**Review by cindyyan317 — APPROVED** [review]

**Pull request merged**